src/main/webapp/app/admin/statistics/statistics-graph.component.ts

```diff
@@ -51,10 +51,6 @@ export class StatisticsGraphComponent implements OnInit, OnChanges {
         this.currentSpan = changes.currentSpan?.currentValue;
         this.barChartLabels = [];
         this.currentPeriod = 0;
-        this.initializeChart();
-    }
-
-    ngOnInit() {
         this.amountOfStudents = this.translateService.instant('statistics.amountOfStudents');
         this.chartName = this.translateService.instant(`statistics.${this.graphType.toString().toLowerCase()}`);
```

Comment:
Those two translations don't change when something else on the page changes, do they? That'd mean we don't need to re-instantiate them every time. On the other hand, re-calculating the translations is probably no real overhead, so feel free to ignore this if you think the current way is good enough.

Reply:
I did not test it but my thought behind it was that language changes should be detected and adapted in the charts. On second thought, a language change might not trigger ngOnChanges since it's not an input, does it?

Reply:
The problem is, that the initializeChart() method in the ngOnChanges need those translations and since this method is executed before ngOnInit(), it is not translated yet.

Reply:
It doesn't seem to make a difference.